src/include/utils/left_factoring.h

```diff
@@ -0,0 +1,13 @@
+#ifndef JUCC_LEFT_FACTORING_H
+#define JUCC_LEFT_FACTORING_H
+#include "grammar/grammar.h"
+namespace grammar = jucc::grammar;
+
+namespace jucc::utils {
+grammar::Productions RemoveLeftFactors(const grammar::Production & /*prod*/);
```

Comment:
Header file function declarations not commented

Reply:
haaat bc

Reply:
ektu likhe the trie.h ta toh aesthetic lagche

Reply:
Done!!